src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -55,84 +51,72 @@ const defaultProps = {
+
+        const errors = {};
+
+        if (_.isEmpty(values.phoneOrEmail)) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.contactMethodRequired');
+        }
+
+        if (!_.isEmpty(values.phoneOrEmail) && !(Str.isValidPhone(phoneLogin) || Str.isValidEmail(values.phoneOrEmail))) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.invalidContactMethod');
+        }
+
+        if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, (values.phoneOrEmail).toLowerCase())) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.enteredMethodIsAlreadySubmited');
+        }
+
+        if (!Permissions.canUsePasswordlessLogins(props.betas) || _.isEmpty(values.password)) {
+            errors.password = props.translate('contacts.genericFailureMessages.passwordRequired');
         }
-        User.addNewContactMethodAndNavigate(login, password);
-    }, [login, props.loginList, password]);
+
+        return errors;
+    };
+
+    const submitForm = (value) => {
+        User.addNewContactMethodAndNavigate(value.phoneOrEmail, value.password);
+    };
 
     return (
-        <ScreenWrapper
-            onTransitionEnd={() => {
-                if (!loginInputRef.current) {
-                    return;
-                }
-                loginInputRef.current.focus();
-            }}
-        >
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
             <HeaderWithCloseButton
                 title={props.translate('contacts.newContactMethod')}
                 shouldShowBackButton
                 onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS)}
                 onCloseButtonPress={() => Navigation.dismissModal(true)}
             />
-            <ScrollView>
-                <Text style={[styles.ph5, styles.mb5]}>
+            <Form
+                formID={ONYXKEYS.FORMS.NEW_CONTACT_FORM}
+                validate={isFormValid}
+                onSubmit={submitForm}
+                submitButtonText={props.translate('common.add')}
+                style={[styles.flexGrow1, styles.mh5]}
+            >
+                <Text style={[styles.mb5]}>
                     {props.translate('common.pleaseEnterEmailOrPhoneNumber')}
                 </Text>
-                <View style={[styles.ph5, styles.mb6]}>
+                <View style={[styles.mb6]}>
                     <TextInput
                         label={`${props.translate('common.email')}/${props.translate('common.phoneNumber')}`}
-                        ref={loginInputRef}
-                        value={login}
-                        onChangeText={handleLoginChange}
+                        inputID="phoneOrEmail"
                         autoCapitalize="none"
                         returnKeyType={Permissions.canUsePasswordlessLogins(props.betas) ? 'done' : 'next'}
+                        autoFocus
```

Comment:
I think it's better to use the previous implementation to focus on the input with `onTransitionEnd`. With this I found it's a bit weird transition with iOS and on Android sometimes the keyboard is not open even the input is focused.

**iOS**


https://user-images.githubusercontent.com/25520267/235652053-26bb0a67-8c00-4d7e-8e12-ea660ee830df.mp4

**Android**


https://user-images.githubusercontent.com/25520267/235652124-b3bd112f-44ba-4210-832e-12f37f2120f4.mov